src/Http/Session/Resolvers/CookieSessionIdResolver.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Http\Session\Resolvers;
+
+use Tempest\Clock\Clock;
+use Tempest\Http\Cookie\CookieManager;
+use Tempest\Http\Session\Session;
+use Tempest\Http\Session\SessionConfig;
+use Tempest\Http\Session\SessionId;
+use Tempest\Http\Session\SessionIdResolver;
+
+final readonly class CookieSessionIdResolver implements SessionIdResolver
+{
+    public function __construct(
+        private CookieManager $cookies,
+        private SessionConfig $sessionConfig,
+        private Clock $clock,
+    ) {
+    }
+
+    public function resolve(): SessionId
+    {
+        $id = $this->cookies->get(Session::ID)?->value ?? null;
+
+        if (! $id) {
+            $id = uniqid();
```

Comment:
I'm thinking we should probably refactor this. There may be some security concerns with this method?